src/builder.rs

```diff
@@ -130,12 +173,33 @@ impl SpendInfo {
         self.note.value()
     }
 
-    fn has_matching_anchor(&self, anchor: Node) -> bool {
+    /// Defined in [Zcash Protocol Spec § 4.8.3: Dummy Notes (Sapling)][orcharddummynotes].
+    ///
+    /// [orcharddummynotes]: https://zips.z.cash/protocol/nu5.pdf#orcharddummynotes
+    fn dummy<R: RngCore>(mut rng: R) -> Self {
+        let (sk, _, note) = Note::dummy(&mut rng);
+        let merkle_path = MerklePath::from_parts(
+            iter::repeat_with(|| Node::from_scalar(jubjub::Base::random(&mut rng)))
+                .take(NOTE_COMMITMENT_TREE_DEPTH.into())
+                .collect(),
+            Position::from(rng.next_u64()),
```

Comment:
To match the protocol spec (`pos`):
```suggestion
            Position::from(0),
```